zio-http/jvm/src/test/scala/zio/http/endpoint/openapi/OpenAPIGenSpec.scala

```diff
@@ -223,6 +226,44 @@ object OpenAPIGenSpec extends ZIOSpecDefault {
       .out[SimpleOutputBody]
       .outError[NotFoundError](Status.NotFound)
 
+  private val apiKeyAuthHeaderEndpoint =
+    Endpoint(GET / "withAuthHeader")
+      .in[SimpleInputBody]
+      .header(HttpCodec.Header.apply(apiKeyHeaderName, StringCodec).optional)
+      .out[SimpleOutputBody]
+      .outError[NotFoundError](Status.NotFound)
+
+  private val apiKeyAuthQueryEndpoint =
+    Endpoint(GET / "withAuthQuery")
+      .in[SimpleInputBody]
+      .query(HttpCodec.query[String](apiKeyQueryParamName).optional)
```

Comment:
This should be handled via `AuthType.Custom`

Reply:
so this should be 
```scala
private val apiKeyAuthQueryEndpoint =
    Endpoint(GET / "withAuthQuery")
      .in[SimpleInputBody]
      .query(HttpCodec.query[String](apiKeyQueryParamName).optional)
      .out[SimpleOutputBody]
      .outError[NotFoundError](Status.NotFound)
      .auth(AuthType.Custom)
 ```
 
 cc @987Nabil ?

Reply:
Yes